### ISSUE TITLE
[BUG] Undefined symbol BGHealthResearchTaskRequest when building app

### ISSUE THREAD

**Issue by frankvollebregt** [transistorsoft/flutter_background_fetch#363]
**Your Environment**
* Plugin version: 1.2.3
* Platform: iOS
* OS version: N/A
* Device manufacturer / model: IPA Archive
* Flutter info (`flutter info`, `flutter doctor`):
```
[✓] Flutter (Channel stable, 3.19.0, on macOS 12.7.1 21G920 darwin-x64, locale nl-NL)
[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
[✓] Chrome - develop for the web
[!] Android Studio (not installed)
[✓] Connected device (3 available)
[✓] Network resources

! Doctor found issues in 2 categories.
```
**To Reproduce**
1. Build your app with version 1.2.2
2. It works fine
3. Update to plugin version 1.2.3 from 1.2.2
4. The build fails with the error below

**Debug logs**
```
$ flutter build ipa

Archiving X.X.X...
Automatically signing iOS for device deployment using specified development team in Xcode project: XXXX
Running Xcode build...                                                  
Xcode archive done.                                         23,3s
Failed to build iOS app
Error (Xcode): Undefined symbol: _OBJC_CLASS_$_BGHealthResearchTaskRequest


Encountered error while archiving for device.
```
**Additional context**
I assume it may have something to do with this other issue: https://github.com/transistorsoft/flutter_background_fetch/issues/357 as it seems to mention this exact class.

A verbose run didn't yield any useful logging from what I could tell. 

I am using this plugin transitively through `flutter_background_geolocation`.

**Comment by christocracy** [comment]
What version of XCode are you using?

**Comment by christocracy** [comment]
>Xcode - develop for iOS and macOS (Xcode 14.2)

If you upgrade to latest XCode, that symbol will be recognized.

**Comment by frankvollebregt** [comment]
Aha, thank you, good to know. I already looked on the App Store if there was an update available (there wasn't one), so I assumed it was something else.

It's a bit odd, in the app store it says _New: Xcode 15.2 includes ..._, but when I start Xcode it says 14.2. I'm not very used to the Mac ecosystem, so I'll see if I can somehow correctly upgrade.

I was actually running into this issue in my CD flow on GitHub Actions initially, where I just use the `macos-latest` runner. I guess that is also yet to be updated... I'll go investigate myself, and close this issue once I've managed to resolve it.

The failed GH Action lists the following:
```
Current runner version: '2.314.1'
Operating System
  macOS
  12.7.3
  21H1015
Runner Image
  Image: macos-12
  Version: 20240218.1
  Included Software: https://github.com/actions/runner-images/blob/macOS-12/20240218.1/images/macos/macos-12-Readme.md
  Image Release: https://github.com/actions/runner-images/releases/tag/macOS-12%2F20240218.1
```
And if I look at the included software page that is linked, it does indeed list [Xcode 14.2](https://github.com/actions/runner-images/blob/macOS-12/20240218.1/images/macos/macos-12-Readme.md#xcode) as its newest included version. Confusing! But not a problem caused by this package I assume.
